kabob-build/src/main/resources/edu/ucdenver/ccp/kabob/build/rules/bio_labels/proteins.clj

```diff
@@ -19,21 +19,58 @@
    (_/record obo/BFO_0000051 _/typeRecordField)
    (_/typeRecordField kiao/hasTemplate iaouniprot/ProteinSchema1)
    (_/typeRecordField rdf/type iaouniprot/Protein)
-   ;; This record will also contain the recommended name of the protein.
-   (_/typeRecordField obo/BFO_0000051 _/recommendedNameRecord)
-   (_/recommendedNameRecord kiao/hasTemplate iaouniprot/RecommendedNameSchema1)
-   (_/recommendedNameRecord rdf/type iaouniprot/RecommendedName)
-   ;; The recommended name field is an evidenced string.
-   (_/recommendedNameRecord obo/BFO_0000051 _/evidencedStringRecord)
-   (_/evidencedStringRecord kiao/hasTemplate iaouniprot/EvidencedStringSchema1)
-   (_/evidencedStringRecord rdf/type iaouniprot/EvidencedString)
-   ;; The value of the evidenced string is what we want as the protein name.
-   (_/evidencedStringRecord obo/BFO_0000051 _/evidencedStringValue)
-   (_/evidencedStringValue obo/IAO_0000219 ?/proteinName))
+   ;; This record will also contain the unique name of the protein.
+   (_/record obo/BFO_0000051 _/nameRecord)
+   (_/nameRecord kiao/hasTemplate iaouniprot/UniProtFileRecord_nameDataField1)
+   (_/nameRecord obo/IAO_0000219 ?/proteinName))
 
  ;; The form of the new triples that will be constructed.
  :head
- '((?/bio rdf/label ?/proteinName))
+ '((?/bio rdfs/label ?/proteinName))
+
+ ;; In this case no `:reify` entry is required, since all of the data required
+ ;; to form the new triples already exist.
+
+ :options
+ {:magic-prefixes
+  [;; AllegroGraph processing directives; http://franz.com/agraph/support/documentation/current/sparql-reference.html#header2-138
+
+   ;; Note: DO NOT set the clause reorderer to `identity`.  The default
```

Comment:
You're absolutely right.  The order makes a big difference if we were to use the `identity` clause reorderer.  But, assuming I'm interpreting the query plan numbers correctly, the `statistical` clause reorderer appears to do a much better job than manually ordering the clauses.  The `cursor-estimate` costs are the same, but the `triple-query`cost is much lower for the `statistical` reorderer.